commit a517c2adce5683483d9e15729b835dffa8d34075
Author: Yang Bo <yangbo@linuxdeepin.com>
Date:   Sun Jan 3 19:38:23 2016 +0800

    UBUNTU: SAUCE: (no-up) x86: reboot: Make Dell Optiplex 390 use reboot=pci
    
    BugLink: http://bugs.launchpad.net/bugs/800660
    
    From: Leann Ogasawara <leann.ogasawara@canonical.com>
    
    Signed-off-by: Leann Ogasawara <leann.ogasawara@canonical.com>
    Signed-off-by: Joseph Salisbury <joseph.salisbury@canonical.com>
    Signed-off-by: Tim Gardner <tim.gardner@canonical.com>
    
    Signed-off-by: Yang Bo <yangbo@linuxdeepin.com>

diff --git a/arch/x86/kernel/reboot.c b/arch/x86/kernel/reboot.c
index 8db0cee..fca04a4 100644
--- a/arch/x86/kernel/reboot.c
+++ b/arch/x86/kernel/reboot.c
@@ -439,6 +439,14 @@ static struct dmi_system_id __initdata reboot_dmi_table[] = {
 			DMI_MATCH(DMI_PRODUCT_NAME, "Latitude E6220"),
 		},
 	},
+	{	/* Handle problems with rebooting on the OptiPlex 390. */
+		.callback = set_pci_reboot,
+		.ident = "Dell OptiPlex 390",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "Dell Inc."),
+			DMI_MATCH(DMI_PRODUCT_NAME, "OptiPlex 390"),
+		},
+	},
 	{ }
 };
 
